### PR TITLE
[TS] LPS-177569 Recent Bloggers portlet crashes with SQLSyntaxErrorException when setting selection method to Scope

### PR DESCRIPTION
**Pull request by IstvanD**
Dear Team,

Could you please review this pull request?

Motivation
=======
When changing the selection method to "Scope" in the Recent Bloggers portlet, it crashes with SQLSyntaxErrorException, because [BlogsStatsUserLocalServiceImpl](https://github.com/liferay/liferay-portal/blob/master/modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsStatsUserLocalServiceImpl.java) produces a syntactically incorrect SQL query.

`select count(distinct BlogsEntry.entryId) entryCount, BlogsEntry.groupId, max(BlogsEntry.modifiedDate) lastPostDate, count(distinct RatingsEntry.entryId) ratingsTotalEntries, avg(RatingsEntry.score) ratingsAverageScore, sum(RatingsEntry.score) ratingsTotalScore, BlogsEntry.userId from BlogsEntry left join RatingsEntry on RatingsEntry.classNameId = ? and RatingsEntry.classPK = BlogsEntry.entryId where BlogsEntry.groupId = ? and entryCount != ? group by BlogsEntry.groupId, BlogsEntry.userId order by lastPostDate desc`

Column aliases, like `entryCount` are created in SELECT and later used in the WHERE clause, but they cannot be referenced because they are applied after the WHERE clause is processed.
https://stackoverflow.com/questions/8370114/referring-to-a-column-alias-in-a-where-clause

Proposed Solution
========
Restructuring the query to place the SELECT in an inline view. This way, SELECT will be handled before the WHERE clause so it can refer to the field aliases.
https://oracle-patches.com/en/databases/sql/referencing-an-aliased-column-in-the-where-clause-via-sql-query

Steps to Verify
========
Please see the linked LPS for the reproduction steps.
https://issues.liferay.com/browse/LPS-177569

Thank you and best regards,
István

### CODE REVIEW

File: modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsStatsUserLocalServiceImpl.java [hunk 1]
```diff
@@ -58,27 +60,23 @@
 	public List<BlogsStatsUser> getGroupsStatsUsers(
 		long companyId, long groupId, int start, int end) {
 
+		Predicate companyIdEquals = _companyIdAlias.eq(companyId);
```
Comment (adolfopa) [inline modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsStatsUserLocalServiceImpl.java:63]: Can you inline this variable? Same with other `*IdEquals` variables.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsStatsUserLocalServiceImpl.java (hunk 1 above)
Comment (IstvanD) [reply modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsStatsUserLocalServiceImpl.java:63]: Hi @adolfopa I originally inlined the expressions, but then the source formatter complained about chaining of the eq() and and() methods.

**Comment by adolfopa** [comment]
ci:forward

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
@IstvanD Test failures look related to your changes. Can you take a look?

**Comment by IstvanD** [comment]
Hi @adolfopa I made some adjustments and force-pushed them. The tests should pass now.

**Comment by austinchiang** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
@IstvanD Looks good. Let's wait for QA to evaluate the test failures.

**Comment by austinchiang** [comment]
LGTM

**Comment by austinchiang** [comment]
ci:forward

**Pull request closed without merging**